src/analysis/interpolation/qgsdualedgetriangulation.cpp

```diff
@@ -1327,13 +1327,39 @@ unsigned int QgsDualEdgeTriangulation::insertEdge( int dual, int next, int point
 
 }
 
+static bool triangleIsFlat( const QgsPoint &pt1, const QgsPoint &pt2, const QgsPoint &pt3, double tolerance )
```

Comment:
Maybe this method could be useful in QgsTriangle class?

Reply:
Or in QgsGeometryUtils to avoid the creation of a triangle each time

Reply:
Ad this function is not complete (but sufficient in this context), no sure. Indeed, only the distance between the segment [pt1,pt2] from pt3 is checked. So if this segment is very small and pt3 far away, the triangle is flat but the function does not return `true`. 
This is acceptable here and changing that will slow down the function. I think I will change the function name / add comments.